pkg/docker/docker-entrypoint.sh

```diff
@@ -1,11 +1,11 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
 
 set -e
 
 curl_put()
 {
-    RET=`/usr/bin/curl -s -w '%{http_code}' -X PUT --data-binary @$1 --unix-socket /var/run/control.unit.sock http://localhost/$2`
-    RET_BODY=${RET::-3}
+    RET=$(/usr/bin/curl -s -w '%{http_code}' -X PUT --data-binary @$1 --unix-socket /var/run/control.unit.sock http://localhost/$2)
+    RET_BODY=$(printf '%s' "$RET" | cut -c 1)
```

Comment:
I don't fully understand bash's `${RET::-3}`. I agree with you it's quite cryptic, and prefer your approach with pipes and standard commands.

But could you please explain why those two are equivalent?  The bash code is removing the last 3 characters of a string, and your code is keeping only the first character.

Please show which kind of string is expected/valid at this point, and why both are equivalent.

Reply:
so I figured what its doing in original file
what I did was clearly nonsense

so what it does is simply get the string without the last 3 characters, because the last 3 chars are the http response code

Reply:
okay I fixed it
I hope its clear now